hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java

```diff
@@ -391,6 +391,9 @@ public static DatanodeDetails.Builder newBuilder(
     }
     if (datanodeDetailsProto.hasCurrentVersion()) {
       builder.setCurrentVersion(datanodeDetailsProto.getCurrentVersion());
+    } else {
+      // fallback to version 1 if not present
```

Comment:
I would suggest to remove this fallback branch since it's not sure about the DN's version.